application/dash_application/views/call_ready_rna.py

```diff
@@ -134,57 +129,64 @@ def get_merged_rna_data():
 RNA_DF = add_graphable_cols(RNA_DF, initial, shape_colour.items_for_df(), None, True)
 
 
-def generate_unique_reads(df, graph_params):
-   return generate(
-       "🚧 Unique Reads (PF) -- DATA MAY BE SUSPECT 🚧", df,
+def generate_total_reads_subplot(df, graph_params):
+    return generate_traces(df,
+        lambda d: d[util.ml_col],
+        lambda d: d[special_cols["Total Reads (Passed Filter)"]],
+        graph_params,
+        [(cutoff_pf_reads_label, initial[cutoff_pf_reads])],
+        util.ml_col, showlegend=True)
+
+
+def generate_unique_reads_subplot(df, graph_params):
+   return generate_traces(df,
        lambda d: d[util.ml_col],
        lambda d: d[special_cols["Unique Reads (PF)"]],
-       "%", graph_params["colour_by"], graph_params["shape_by"],
-       graph_params["shownames_val"], [],
-       util.ml_col)
+       graph_params, [], util.ml_col)
 
 
-def generate_five_to_three(df, graph_params):
-    return generate(
-       "5 to 3 Prime Bias", df,
+def generate_five_to_three_subplot(df, graph_params):
+    return generate_traces(df,
        lambda d: d[util.ml_col],
        lambda d: d[RNASEQQC2_COL.MetricsMedian5PrimeTo3PrimeBias],
-       "", graph_params["colour_by"], graph_params["shape_by"],
-       graph_params["shownames_val"], [],
-       util.ml_col)
+       graph_params, [], util.ml_col)
 
 
-def generate_correct_read_strand(df, graph_params):
-    return generate(
-        "🚧 % Correct Read Strand -- DATA MAY BE SUSPECT 🚧", df,
+def generate_correct_read_strand_subplot(df, graph_params):
+    return generate_traces(df,
         lambda d: d[util.ml_col],
         lambda d: d[RNASEQQC2_COL.MetricsPercentCorrectStrandReads],
-        "%",graph_params["colour_by"], graph_params["shape_by"],
-        graph_params["shownames_val"], [],
-        util.ml_col)
+        graph_params, [], util.ml_col)
 
 
-def generate_coding(df, graph_params):
-    return generate(
-        "% Coding", df,
+def generate_coding_subplot(df, graph_params):
+    return generate_traces(df,
         lambda d: d[util.ml_col],
         lambda d: d[RNASEQQC2_COL.MetricsPercentCodingBases],
-        "%", graph_params["colour_by"], graph_params["shape_by"],
-        graph_params["shownames_val"], [],
-        util.ml_col)
+        graph_params, [], util.ml_col)
 
 
-def generate_rrna_contam(df, graph_params):
-    return generate(
-        "🚧 % rRNA Contamination -- DATA MAY BE SUSPECT 🚧", df,
+def generate_rrna_contam_subplot(df, graph_params):
+    return generate_traces(df,
         lambda d: d[util.ml_col],
         lambda d: d[special_cols["% rRNA Contamination"]],
-        "%", graph_params["colour_by"], graph_params["shape_by"],
-        graph_params["shownames_val"],
+        graph_params,
         [(cutoff_rrna_contam_label, graph_params[cutoff_rrna_contam])],
         util.ml_col)
 
 
+graphs = [
```

Comment:
Is there any way to keep the axis labels inside the generate_*_subplot functions? To me this seems like having to look in multiple places for the unchanging parts of each graph, when before everything that wasn't affected by the sidebar was contained in the function

Reply:
(I'm actively working on this because it's a good idea, but it's taking a bit longer than I expected!)

Reply:
Refactored to add "Subplot" classes which store the graph traces, title, and y_label; let me know what you think!